packages/eslint-plugin/src/rules/no-misused-promises.ts

```diff
@@ -944,18 +966,44 @@ function getHeritageTypes(
     .map(typeExpression => checker.getTypeAtLocation(typeExpression));
 }
 
+function getWellKnownStringOfSymbol(symbol: symbol): string | null {
+  switch (symbol) {
+    case Symbol.iterator:
+      return 'iterator';
+    case Symbol.asyncIterator:
+      return 'asyncIterator';
+  }
+
+  return null;
+}
```

Comment:
I'm not sure if there's a better way to map a symbol to its string counterpart `tsutils.getWellKnownSymbolPropertyOfType` is expecting. The list for [well-known symbols](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol#well-known_symbols) isn't long, but this will still require maintaining this mapping.

Reply:
If this approach makes sense, this list can be expanded to support other well-known symbols.

Reply:
Is there a reason we need to stick to well-known symbols? I'd think we should go with any detected symbol, no?

Reply:
Hmm, I might be missing something, but I'm not entirely sure how to check if a JavaScript `symbol` exists on another `ts.Type` 🤔 

Here's how existing code checks if an inherited type has a `string` symbol:

```ts
const escapedMemberName = ts.escapeLeadingUnderscores(staticAccessValue);
const symbolMemberMatch = type.getSymbol()?.members?.get(escapedMemberName);

return (
  symbolMemberMatch ?? tsutils.getPropertyOfType(type, escapedMemberName)
);
```

I've looked into the `SymbolObject` but couldn't find a way to check against an arbitrary `symbol` (not to be confused with `ts.Symbol`).

Because of this, I've created a function to map between known symbols and their string representation, which are passed to `tsutils.getWellKnownSymbolPropertyOfType` to check if they exist on the inherited type.

Reply:
Hmm gotcha thanks. cc @kirkwaiblinger - I'm fine with this as a patchwork helper, with as little hardcoding as possible.

```ts
function getWellKnownStringOfSymbol(symbol: symbol): string | null {
  for (const [key, value] of Object.entries(Symbol)) {
    if (symbol === value) {
      return key;
    }
  }

  return null;
}
```

And, sorry for dropping the ball on this again 🤦 there's something about symbols that makes me instinctively forget the conversation...

Reply:
I've updated the PR to not hard-code well-known symbols, thanks!

The only alternative I had in mind for not hard-coding was to extract the value from [`Symbol.prototype.toString()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Symbol/toString) with a regex (which I really didn't want to do), but I didn't think of iterating the global `Symbol`'s keys.

This means that arbitrary symbols (e.g. `Symbol.for('foo')`) won't be reported by the rule though.

Reply:
Yeah I don't see a way to work for arbitrary symbols 😞

Reply:
cross-linking discord thread https://discord.com/channels/1026804805894672454/1322853884292497418